x/concentrated-liquidity/internal/model/pool.go

```diff
@@ -182,3 +182,25 @@ func (p *Pool) ApplySwap(newLiquidity sdk.Dec, newCurrentTick sdk.Int, newCurren
 	p.CurrentSqrtPrice = newCurrentSqrtPrice
 	return nil
 }
+
+// TODO: Figure out what we are going to do with these stubs since we call these in the keeper but are needed to use PoolI
+func (p Pool) CalcInAmtGivenOut(
+	ctx sdk.Context, tokensOut sdk.Coins, tokenInDenom string, swapFee sdk.Dec) (tokenIn sdk.Coin, err error) {
+	return sdk.Coin{}, nil
+}
+
+func (p Pool) CalcOutAmtGivenIn(ctx sdk.Context, tokensIn sdk.Coins, tokenOutDenom string, swapFee sdk.Dec) (sdk.Coin, error) {
+	return sdk.Coin{}, nil
+}
```

Comment:
Why? We shouldn't need these as they should be defined on the `TraditionalAmmInterface`

Reply:
They are defined in PoolI within swaprouter here, not TraditionalAmmInterface

https://github.com/osmosis-labs/osmosis/blob/dba9df0db5a8330de4204e52ec0369cea31c570c/x/swaprouter/types/pool.go#L30-L42

Reply:
Ahh I see, these should be defined on the `TraditionalAmmInterface`. I'll try to see if moving these is a quick change. If so, maybe we can merge that change first to avoid creating the stubs here unnecessarily

Reply:
Didn't we have these defined on TAMM previously and then decided to move them up to PoolI?

Reply:
No, based on the component diagram, the goal has been to have it on `TraditionalAmmInterface`. The only reason why we would keep these methods on `PoolI` until now it is to be able to port `EstimateSwap` to swaprouter. However, since we now call keeper's "Calc" method instead of the pool's, we can move these to the right place.

I've looked into this problem, and moving the methods to `TraditionalAmmInterface` depends on this PR. Specificcally, on the change I mentioned above (where swaprouter uses keeper's calc as opposed to pool's).

So I think we can merge this first and refactor separately.